lib/engine/game/g_1849/game.rb

```diff
@@ -771,12 +824,137 @@ def bank_sort(corporations)
         end
 
         def player_value(player)
-          player.value - @player_debts[player]
+          player.value - @player_debts[player] - player.shares_by_corporation.sum do |corp, _|
+            player.num_shares_of(corp) * corp.loans.size * 100
+          end
         end
 
         def par_prices
           @stock_market.share_prices_with_types(@available_par_groups)
         end
+
+        # code below is for variant rules
+
+        def acquiring_station_tokens?
+          @acquiring_station_tokens ||= @optional_rules&.include?(:acquiring_station_tokens)
+        end
+
+        def electric_dreams?
+          @electric_dreams ||= @optional_rules&.include?(:electric_dreams)
+        end
+
+        def bonds?
+          @bonds ||= @optional_rules&.include?(:bonds)
+        end
+
+        def reduced_4p_corps?
+          @reduced_4p_corps ||= @optional_rules&.include?(:reduced_4p_corps)
+        end
+
+        def event_buy_tokens!
+          @log << "-- Event: #{EVENTS_TEXT[:buy_tokens][1]} --"
+        end
+
+        def event_bonds!
+          @log << "-- Event: #{EVENTS_TEXT[:bonds][1]} --"
+        end
+
+        def event_e_tokens!
+          @log << "-- Event: #{EVENTS_TEXT[:e_tokens][1]} --"
+        end
+
+        # code below is for Bonds variant
+        NUM_LOANS = 6
+        MAXIMUM_LOANS = 1
+        LOAN_VALUE = 500
+        INTEREST_RATE = 50
+
+        def init_loans
+          Array.new(num_loans) { |id| Loan.new(id, loan_value) }
+        end
+
+        def num_loans
+          bonds? ? NUM_LOANS : 0
+        end
+
+        def maximum_loans(_entity)
+          MAXIMUM_LOANS
+        end
+
+        def loan_value(_entity = nil)
+          LOAN_VALUE
+        end
+
+        def interest_rate
+          INTEREST_RATE
+        end
+
+        def interest_owed_for_loans(loans)
+          interest_rate * loans
+        end
+
+        def interest_owed(entity)
+          interest_owed_for_loans(entity.loans.size)
+        end
+
+        def loans_due_interest(entity)
+          entity.loans.size
+        end
+
+        def log_interest_payment(entity, amount)
+          amount_fmt = format_currency(amount)
+          @log << "#{entity.name} pays #{amount_fmt} interest for its issued bond"
+        end
+
+        def take_loan(entity, loan)
+          raise GameError, 'Cannot issue bond' unless can_take_loan?(entity)
+
+          name = entity.name
+          @log << "#{name} issues its bond and receives #{format_currency(loan_value)}"
+          @bank.spend(loan_value, entity)
+          entity.loans << loan
```

Comment:
Just create a loan here. No need to have an array of them.

Reply:
You can still give the entity a loan. It just doesn't have to come from a common loan pool. It can be a new loan each time. Similar to how infinite tile logic works.

Reply:
done